src/karts/cached_characteristic.cpp

```diff
@@ -0,0 +1,203 @@
+//
+//  SuperTuxKart - a fun racing game with go-kart
+//  Copyright (C) 2006-2015 SuperTuxKart-Team
+//
+//  This program is free software; you can redistribute it and/or
+//  modify it under the terms of the GNU General Public License
+//  as published by the Free Software Foundation; either version 3
+//  of the License, or (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+
+#include "karts/cached_characteristic.hpp"
+
+#include "utils/interpolation_array.hpp"
+
+CachedCharacteristic::CachedCharacteristic(const AbstractCharacteristic *origin) :
+    m_values(CHARACTERISTIC_COUNT),
+    m_origin(origin)
+{
+    updateSource();
+}
+
+// ----------------------------------------------------------------------------
+/** Deletes all allocated values. */
+CachedCharacteristic::~CachedCharacteristic()
+{
+    // Delete all not-null values
+    for (int i = 0; i < CHARACTERISTIC_COUNT; i++)
+    {
+        SaveValue &v = m_values[i];
+        if (v.content)
+        {
+            switch (getType(static_cast<CharacteristicType>(i)))
+            {
+            case TYPE_FLOAT:
+                delete static_cast<float*>(v.content);
+                break;
+            case TYPE_FLOAT_VECTOR:
+                delete static_cast<std::vector<float>*>(v.content);
+                break;
+            case TYPE_INTERPOLATION_ARRAY:
+                delete static_cast<InterpolationArray*>(v.content);
+                break;
+            case TYPE_BOOL:
+                delete static_cast<bool*>(v.content);
+                break;
+            }
+            v.content = nullptr;
+        }
```

Comment:
This works well, though I'm wondering, couldn't the destructor of Value handle that? Seems like it would belong there ideally